src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs

```diff
@@ -150,6 +153,19 @@ private static bool GetIsLinux()
                 return true;
             }
 
+            try
```

Comment:
Should we special case netcoreapp and use the `IsOSPlatform` method here?

Reply:
I'd say yes, especially given that `libSystem.Native` appears to be a .NET (Core) library. There's a catch all, so it won't crash NuGet, but it also means that this (probably) won't work on Mono on FreeBSD

Reply:
We use following fragment in runtime in places that needs to compile for netstandard.  
```c# 
if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")))
{

```

Reply:
BTW for reference old msbuild change: https://github.com/dotnet/msbuild/pull/2539/files

Reply:
I've adopted your suggestion. Thank you, @wfurt.

> Someone needs to run some tests on Mono on FreeBSD

@zivkan
I've run the following code with `mono` and `dotnet` on FreeBSD:

```cs
using System;
using System.Runtime.InteropServices;

namespace ConsoleApp
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Type.GetType(\"Mono.Runtime\") != null: {0}", Type.GetType("Mono.Runtime") != null);
            Console.WriteLine("RuntimeInformation.IsOSPlatform(OSPlatform.Create(\"FreeBSD\")): {0}", RuntimeInformation.IsOSPlatform(OSPlatform.Create("FreeBSD")));
            Console.WriteLine("RuntimeInformation.IsOSPlatform(OSPlatform.Linux): {0}", RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
            Console.WriteLine("Environment.OSVersion.Platform == PlatformID.Unix: {0}", Environment.OSVersion.Platform == PlatformID.Unix);
        }
    }
}
```

These are the results:
- `mono` (`TargetFrameworkVersion`: `v4.8`)
  ```
  Type.GetType("Mono.Runtime") != null: True
  RuntimeInformation.IsOSPlatform(OSPlatform.Create("FreeBSD")): False
  RuntimeInformation.IsOSPlatform(OSPlatform.Linux): True
  Environment.OSVersion.Platform == PlatformID.Unix: True
  ```
- `dotnet` (`TargetFramework`: `net6.0`)
  ```
  Type.GetType("Mono.Runtime") != null: False
  RuntimeInformation.IsOSPlatform(OSPlatform.Create("FreeBSD")): True
  RuntimeInformation.IsOSPlatform(OSPlatform.Linux): False
  Environment.OSVersion.Platform == PlatformID.Unix: True
  ```

Although `RuntimeInformation.IsOSPlatform(OSPlatform.Create("FreeBSD"))` returns `False` when run with `mono`, this shouldn't be a problem because in this case `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)` returns `True` instead.